File: kinesis-data-streams-java/src/main/java/com/github/antontucek/kinesis/data/streams/java/OrderGenerator.java
package com.github.antontucek.kinesis.data.streams.java;

import java.math.BigDecimal;
import java.math.RoundingMode;
import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class OrderGenerator {

    private final Random RANDOM = new Random();
    private final static String[] SKUS = {"Boat", "Jet", "Horse", "Truck", "Van"};
    private final static String[] COUNTRIES = {"CZ", "DE", "GB", "US"};
    private final static  String[] CUSTOMERS = {"Brian", "Fred", "James", "Jane", "Katie", "Megan", "Paige", "Richard"};


    /**
     * Generates random order with 1-3 lines
     * Probability:
     *    - 1 lines: 70%
     *    - 2 lines: 20%
     *    - 3 lines: 10%
     * @return Random order
     */
    public Order getRandomOrder() {
        // Generate OrderItems
        int randomNumber = RANDOM.nextInt(10);
        int lines;
        if (randomNumber < 7) {
            lines = 1;
        } else if (randomNumber < 9) {
            lines = 2;
        } else {
            lines = 3;
        }
        List<OrderItem> orderItemList = new ArrayList<>();
        for (int i = 0; i < lines; i++) {
            orderItemList.add(new OrderItem(SKUS[RANDOM.nextInt(SKUS.length)],
                    round((float) (10.0 + 10.0 * RANDOM.nextFloat()), 2),
                    1 + RANDOM.nextInt(5)));
        }

        // Return
        return new Order(Instant.now(),
                COUNTRIES[RANDOM.nextInt(COUNTRIES.length)],
                CUSTOMERS[RANDOM.nextInt(CUSTOMERS.length)],
                orderItemList);
    }

    /**
     * Rounds float to required decimal place
     * @param f Float value to round
     * @param decimalPlace How many decimal places to round
     * @return Rounded float value
     */
    public static float round(float f, int decimalPlace) {
        return BigDecimal.valueOf(f).setScale(decimalPlace, RoundingMode.HALF_UP).floatValue();
    }
}

File: kinesis-data-streams-kotlin/src/main/kotlin/com/github/antontucek/kinesis/data/streams/kotlin/OrderGenerator.kt
package com.github.antontucek.kinesis.data.streams.kotlin

import java.math.BigDecimal
import java.math.RoundingMode
import java.time.Instant
import java.util.Random

class OrderGenerator {

    private val rg = Random()

    companion object {
        private val SKUS = arrayOf("Boat", "Jet", "Horse", "Truck", "Van")
        private val COUNTRIES = arrayOf("CZ", "DE", "GB", "US")
        private val CUSTOMERS = arrayOf("Brian", "Fred", "James", "Jane", "Katie", "Megan", "Paige", "Richard")
        fun round(f: Float, decimalPlace: Int) = BigDecimal.valueOf(f.toDouble()).setScale(decimalPlace, RoundingMode.HALF_UP).toFloat()
    }

    /**
     * Generates random order with 1-3 lines
     * Probability:
     *    - 1 lines: 70%
     *    - 2 lines: 20%
     *    - 3 lines: 10%
     * @return Random order
     */
    fun randomOrder(): Order {
        val randomNumber = rg.nextInt(10)
        val lines = when {
            (randomNumber < 7) -> 1
            (randomNumber < 9) -> 2
            else -> 3
        }
        val orderItemList = mutableListOf<OrderItem>()
        for (i in 0 until lines) {
            orderItemList.add(OrderItem(SKUS[rg.nextInt(SKUS.size)], round((10.0 + 10.0 * rg.nextFloat()).toFloat(), 2), 1 + rg.nextInt(5)))
        }
        return Order(Instant.now(), COUNTRIES[rg.nextInt(COUNTRIES.size)], CUSTOMERS[rg.nextInt(CUSTOMERS.size)], orderItemList)
    }

}